client/src/components/CategoriesScreen.tsx

```diff
@@ -44,18 +46,18 @@ export default function CategoriesScreen({
           <ModalNewCategory
             modalAddVisible={modalAddVisible}
             setModalAddVisible={setModalAddVisible}
-            setCategories={setCategories}
-            categories={categories}
+            userId={user._id}
```

Comment:
Better to use separate state handler mechanism to avoid passing/removing user for any refactoring from props and types